pennylane/transforms/draw.py

```diff
@@ -152,6 +152,148 @@ def wrapper(*args, **kwargs):
+
+    >>> print(qml.draw(circuit)(a=2.3, w=[1.2, 3.2, 0.7]))
+    0: ──H─╭C───────╭C──┤ ╭<Z@Z>
+    1: ────╰RX──Rot─╰RX─┤ ╰<Z@Z>
+
+    .. UsageDetails::
+
+
+    By default, parameters are omitted. By specifying the ``decimals`` keyword, parameters
+    are displayed to the specified precision. Matrix-valued parameters are never displayed.
+
+    >>> print(qml.draw(circuit, decimals=2)(a=2.3, w=[1.2, 3.2, 0.7]))
+    0: ──H─╭C─────────────────────────────╭C─────────┤ ╭<Z@Z>
+    1: ────╰RX(2.30)──Rot(1.20,3.20,0.70)─╰RX(-2.30)─┤ ╰<Z@Z>
+
+    The ``max_length`` keyword warps long circuits:
+
+    .. code-block:: python
+
+        rng = np.random.default_rng(seed=42)
+        shape = qml.StronglyEntanglingLayers.shape(n_wires=3, n_layers=5)
+        params = rng.random(shape)
+
+        @qml.qnode(qml.device('lightning.qubit', wires=3))
+        def longer_circuit(params):
+            qml.StronglyEntanglingLayers(params, wires=range(3))
+            return [qml.expval(qml.PauliZ(i)) for i in range(3)]
+
+        print(qml.draw(longer_circuit, max_length=60)(params))
+
+    .. code-block:: none
+
+        0: ──Rot─╭C────╭X──Rot─╭C─╭X──Rot──────╭C────╭X──Rot─╭C─╭X
+        1: ──Rot─╰X─╭C─│───Rot─│──╰C─╭X────Rot─╰X─╭C─│───Rot─│──╰C
+        2: ──Rot────╰X─╰C──Rot─╰X────╰C────Rot────╰X─╰C──Rot─╰X───
+
+        ───Rot──────╭C────╭X─┤  <Z>
+        ──╭X────Rot─╰X─╭C─│──┤  <Z>
+        ──╰C────Rot────╰X─╰C─┤  <Z>
+
+    The ``wire_order`` keyword specifies the order of the wires from
+    top to bottom:
+
+    >>> print(qml.draw(circuit, wire_order=[1,0])(a=2.3, w=[1.2, 3.2, 0.7]))
+    1: ────╭RX──Rot─╭RX─┤ ╭<Z@Z>
+    0: ──H─╰C───────╰C──┤ ╰<Z@Z>
+
+    If the device or ``wire_order`` has wires not used by operations, those wires are omitted
+    unless requested with ``show_all_wires=True``
+
+    >>> @qml.qnode(qml.device('lightning.qubit', wires=3))
+        def empty_circuit():
+            return qml.expval(qml.PauliZ(0))
+    >>> print(qml.draw(empty_circuit, show_all_wires=True)())
+    0: ───┤  <Z>
+    1: ───┤
+    2: ───┤
+
+    """
+
+    @wraps(qnode)
+    def wrapper(*args, **kwargs):
+
+        original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
+
+        try:
+            qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
+            tapes = qnode.construct(args, kwargs)
+        finally:
+            qnode.expansion_strategy = original_expansion_strategy
+
+        _wire_order = wire_order or qnode.device.wires
+
+        if tapes is not None:
+            cache = {"tape_offset": 0}
+            res = [
+                tape_text(
+                    t,
+                    wire_order=_wire_order,
+                    show_all_wires=show_all_wires,
+                    decimals=decimals,
+                    max_length=max_length,
+                    cache=cache,
+                )
+                for t in tapes[0]
+            ]

```

Comment:
Out of curiosity what scenario is this intended for?

Reply:
batch transforms.

Reply:
Oh that's awesome, I didn't realise either that the previous drawer could do that as well. Do you think it would be a worthwhile example for the docstring?